src/Components/Endpoints/src/FormMapping/FormDataReader.cs

```diff
@@ -73,6 +73,13 @@ public void AddMappingError(Exception exception, string? attemptedValue)
     {
         ArgumentNullException.ThrowIfNull(exception);
 
+        // Avoid re-wrapping the exception if it is already a FormDataMappingException
+        // and we don't have an ErrorHandler configured.
+        if (exception is FormDataMappingException && ErrorHandler == null)
```

Comment:
Is there a test that exercises this?

Reply:
The `SupportsRecursivePropertiesWithRecursionLimit` test exercises this.

Reply:
Ah a double test, I was only looking at it from the limit applying perspective 😄